src/Screencast/main.vala

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg libportal-gtk4 --pkg gstreamer-1.0
+
+private Gtk.Picture output;
+
+async void start_screencast_session (Xdp.Portal portal, Xdp.Parent parent) {
```

Comment:
Throw an error from this method and handle it in the callback where it is called, just so the entire method body is not inside a try catch statement.

Reply:
Added a commit, let me know if this is the correct implementation